src/lib/components/TimelineView.svelte

```diff
@@ -146,6 +146,41 @@
       (authors) => user.agent && authors.push(user.agent.assertDid),
     );
     message.bodyJson = JSON.stringify(messageInput);
+    let messageJSON = JSON.parse(message.bodyJson) as JSONContent;
+
+    messageJSON.content = (messageJSON.content ?? []).map((block) => {
+      if (block.type === "paragraph" && Array.isArray(block.content)) {
+        let foundText = false;
+        block.content = block.content.filter((inline) => {
+          if (inline.type === "hardBreak" && !foundText) {
+            return false; // remove leading hardBreaks
+          }
+
+          if (inline.type === "text") {
+            inline.text = (inline?.text ?? "").replace(/^\s+/, ""); // remove leading spaces
+            if (inline.text.trim() === "") {
+              return false; // remove text node if it's empty or all spaces
+            }
+            foundText = true;
+          }
+
+          return true;
+        });
+      }
+      return block;
+    });
+
+    console.log("messageJSON", messageJSON);
+
+    // If message is empty, don't save it
+    if (
+      messageJSON.content[0]?.content?.length === 0 ||
+      !messageJSON.content[0]?.content
+    ) {
+      toast.error("Message cannot be empty", { position: "bottom-end" });
```

Comment:
Let's have this just show up in the default location.

```suggestion
      toast.error("Message cannot be empty");
```